### PR TITLE
Allow to use wallet provider

### PR DESCRIPTION
**Pull request by nocmar**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  ],
+  "dependencies": {
+    "@material-ui/core": "^4.11.4",
+    "@material-ui/icons": "^4.11.2",
+    "@terra-money/terra.js": "^1.8.1",
+    "@terra-money/wallet-provider": "^0.15.0",
+    "@types/react-router-dom": "^5.1.7",
+    "@types/uuid": "^8.3.0",
+    "react-router-dom": "^5.2.0",
+    "react-use-clipboard": "^1.0.7",
```
Comment (Mskalba) [inline package.json:74]: Probably for this, for material-ui and for styled components and uuid. We should use:

peerDependencies
Peer dependencies are a special type of dependency that would only ever come up if you were publishing your own package.

Having a peer dependency means that your package needs a dependency that is the same exact dependency as the person installing your package. This is useful for packages like react that need to have a single copy of react-dom that is also used by the person installing it.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    "dist"
+  ],
+  "dependencies": {
+    "@material-ui/core": "^4.11.4",
+    "@material-ui/icons": "^4.11.2",
+    "@terra-money/terra.js": "^1.8.1",
+    "@terra-money/wallet-provider": "^0.15.0",
+    "@types/react-router-dom": "^5.1.7",
+    "@types/uuid": "^8.3.0",
+    "react-router-dom": "^5.2.0",
```
[the remark this answers is not in the document]
Comment (nocmar) [reply package.json:73]: You are right but unfortunately this library wallet-provider is using it and they didn't add this as dependency, the same is with styled component.

File: README.md [hunk 3]
```diff
@@ -1,7 +1,38 @@
...
+```bash
+npm install --save starterra-tool-dapp
+```
+
+## Usage
+
+```tsx
+import React, { Component } from 'react'
+
+import MyComponent from 'starterra-tool-dapp'
```
[the remark this answers is not in the document]
Comment (nocmar) [reply README.md:18]: It was just a default one, I will update it

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (nocmar) [reply package.json:73]: but maybe they use it as peer dependency and it shouldn't be there, what do you think?

File: src/ConnectWallet.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+  const connectWallet = useCallback(() => {
+    if (availableConnectTypes.length > 1) {
+      setShowOptions(true)
+    } else if (availableConnectTypes.length === 1) {
+      connect(availableConnectTypes[0])
+    }
+  }, [availableConnectTypes, connect])
+
+  const disconnectWallet = () => {
```
[the remark this answers is not in the document]
Comment (nocmar) [reply src/ConnectWallet.tsx]: ok done

File: src/ConnectWallet.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    disconnect()
+    setShowContent(false)
+  }
+
+  const onClickAway = () => {
```
[the remark this answers is not in the document]
Comment (nocmar) [reply src/ConnectWallet.tsx]: ok done

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (nocmar) [reply package.json]: Forgot about previous comment :) I added it into devdependency to be able to build and test the library, not sure if it should be in peer dependency, what do you think? as it is probably peer dependency of wallet-provider

File: package.json (hunk 1 above)
Comment (nocmar) [reply package.json]: I moved them to peer dependency and added to dev to be able to build and test this package, but I'm not sure if we should treat uuid as peer dependency ?

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (Mskalba) [reply package.json]: Hmm, I guess we do not need to add it to peer then. If it is peer at wallet-provider. Needs to be checked. Lets maybe try to install it to FE app as soon as we will have this one on npm and we will see if it is needed as peer as well.

File: package.json (hunk 1 above)
Comment (Mskalba) [reply package.json]: it is peer dependency i guess.

**Review by Mskalba — APPROVED** [review]

**Pull request merged**